src/libraries/System.Private.Xml/src/System/Xml/XPath/XPathNavigatorReader.cs

```diff
@@ -1270,7 +1271,7 @@ public override bool MoveToId(string id)
 
         public override string GetAttribute(string localName, string namespaceName)
         {
-            return null;
+            return null!;
```

Comment:
not sure about here but similar API in XmlReader returns nullable

Reply:
I assumed that an instance of XmlEmptyNavigator would never call `GetAttribute` nor `GetNamespace` methods but I could be wrong.
I can change this to have a `Debug.Fail` to prove such point.

Reply:
Base class returns nullable and since we don't care about this since it's not expected to be ever called I think it's better to just have less `!` (so IMO make return value and `namespaceName` nullable)

Reply:
I don't see the base method `XPathNavigator.GetAttribute` returning nullable, are we referring to the same class?
Also, this is public API, I would prefer to keep it as is and just add Debug.Fail, I envision that this may change to `throw new InvalidOperationException()` in the future if we can prove my assumption.